pkg/skaffold/build/cluster/kaniko.go

```diff
@@ -181,3 +186,27 @@ func generateEnvFromImage(imageStr string) ([]v1.EnvVar, error) {
 	generatedEnvs = append(generatedEnvs, v1.EnvVar{Name: "IMAGE_TAG", Value: imgRef.Tag})
 	return generatedEnvs, nil
 }
+
+
+func (b *Builder) createSecretIfRequired(ctx context.Context, buildCtx *io.PipeReader, podName string) error {
+	homeDir, err := homedir.Dir()
+	if err != nil {
+		return err
+	}
+	gacCredentialsFile := filepath.Join([]string{homeDir, ".config", "gcloud", "application_default_credentials.json"}...)
```

Comment:
That was a bit cryptic: I thought the gcloud team wanted to avoid peering into ~/.config/gcloud, but I can't find any supporting documentation, and there's nothing on `gcloud auth application-default` that provides this file.